### PR TITLE
AC-2724: Update rector dependency to the latest version

### PR DESCRIPTION
**Pull request by eliseacornejo**
This PR is to resolve this issue: [AC-2724](https://jira.corp.magento.com/browse/AC-2724). 

This is the PR in rector with all the explanation of why this rule was moved to other place:
https://github.com/rectorphp/rector-src/pull/1814/files

### CODE REVIEW

**Review by loginesta — APPROVED** [review]

File: rector.php [hunk 1]
```diff
@@ -8,7 +8,7 @@
 use Rector\Core\ValueObject\PhpVersion;
 use Rector\Php80\Rector\Class_\StringableForToStringRector;
 use Rector\Php80\Rector\ClassMethod\FinalPrivateToPrivateVisibilityRector;
-use Rector\Php80\Rector\ClassMethod\OptionalParametersAfterRequiredRector;
+use Rector\CodeQuality\Rector\ClassMethod\OptionalParametersAfterRequiredRector;
```
Comment (svera) [inline rector.php:11]: @eliseacornejo will this change work if by any chance rector 0.12.4 is installed? I would upgrade the minimum rector version specified in `composer.json` to `^0.12.17`

Comment (eliseacornejo) [reply rector.php:11]: Yes, you are totally right, this is not going to work on 0.12.4 I will do the change!

**Review by svera — APPROVED** [review]

**Comment by eliseacornejo** [comment]
@magento import pr to magento-commerce/magento-coding-standard

**Comment by m2-github-services** [comment]
@eliseacornejo the Pull Request is successfully imported.

**Pull request merged**